spec/requests/api/authentication_spec.rb

```diff
@@ -292,4 +292,62 @@ def systoken(server_guid, userid, timestamp)
       expect_result_to_have_keys(ENTRYPOINT_KEYS)
     end
   end
+
+  context "Role Based Authorization" do
+    before do
+      FactoryGirl.create(:vm_vmware, :name => "vm1")
+    end
+
+    def vms_get_test(new_identifier)
+      config = Api::ApiConfig.collections.vms.collection_actions.get.first
+      saved_identifier = config.identifier
+      config.identifier = new_identifier
+      yield
+      config.identifier = saved_identifier
```

Comment:
A stub method would read better as well.  It's not clear (perhaps because of the vms_get_test method name), that what you are testing is different combinations of RBAC identifiers via the settings.

compare:

```ruby
vms_get_test("vm_view_role1") do
  #stuff
end
```

vs

```ruby
stub_api_settings(:collections, :vms, :collection, :actions, :get, "vm_view_role1")
#stuff
```